src/tp/tp_reassembler.cpp

```
@@ -247,36 +276,40 @@ bool TpReassembler::add_segment_to_buffer(TpReassemblyBuffer& buffer, const TpSe
 bool TpReassembler::is_reassembling(uint32_t message_id) const {
     platform::ScopedLock const lock(buffers_mutex_);
-    return reassembly_buffers_.find(message_id) != reassembly_buffers_.end();
+    for (const auto& pair : reassembly_buffers_) {
+        if (pair.first.message_id == message_id) {
+            return true;
+        }
+    }
+    return false;
 }
 
 bool TpReassembler::get_reassembly_progress(uint32_t message_id, uint32_t& received_bytes, uint32_t& total_bytes) const {
```

Cppcheck (2.21.0)

[style] The function 'get_reassembly_progress' is never used.

(unusedFunction)